### PR TITLE
chore: update multiple linting packages

### PR DESCRIPTION
**Pull request by Sembauke**
- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/src/components/helpers/loader.test.tsx [hunk 1]
```diff
@@ -1,11 +1,9 @@
-import { render, cleanup, screen } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 import React from 'react';
 
 import Loader from './loader';
 
 describe('<Loader />', () => {
-  afterEach(cleanup);
```
Comment (Sembauke) [inline client/src/components/helpers/loader.test.tsx:7]: `cleanup` is performed automatically by your test runner, you don't need manual cleanups

File: client/src/components/settings/__snapshots__/honesty.test.tsx.snap [hunk 2]
```diff
@@ -1,117 +1,135 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (Sembauke) [inline client/src/components/settings/__snapshots__/honesty.test.tsx.snap:1]: very uncomfortable with this change, but it seems to pass the tests?

File: client/src/components/settings/honesty.test.tsx [hunk 3]
```diff
@@ -1,40 +1,27 @@
-import { Button } from '@freecodecamp/react-bootstrap';
```
Comment (Sembauke) [inline client/src/components/settings/honesty.test.tsx:1]: changed this test to use `@testing-library/react` which resolves all linting errors

File: client/src/components/Intro/intro.test.tsx [hunk 4]
```diff
@@ -1,55 +1,28 @@
 import React from 'react';
```
Comment (Sembauke) [inline client/src/components/Intro/intro.test.tsx:1]: changed this test to use @testing-library/react which resolves all linting errors

File: client/src/components/Intro/intro.test.tsx [hunk 5]
```diff
@@ -1,55 +1,27 @@
...
-    expect(container.findAllByType('blockquote').length === 0).toBeTruthy();
-
-    /**
-     * This rules had to be disabled because the new lint rules are throwing false positives here.
-     * They were interpreting react-test-renderer functions as @testing-library/react functions.
-     */
-    // eslint-disable-next-line testing-library/await-async-query
-    expect(container.findAllByType('h1').length === 1).toBeTruthy();
+    renderWithRedux(<Intro {...loggedOutProps} />);
+    expect(screen.queryByTestId('quote-block')).not.toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (Sembauke) [reply client/src/components/Intro/intro.test.tsx:17]: I have tried doing this multiple times. But it seems to fail every time.

File: package.json [hunk 6]
```diff
@@ -127,13 +127,13 @@
     "eslint-import-resolver-typescript": "^3.5.5",
     "eslint-plugin-filenames-simple": "0.9.0",
     "eslint-plugin-import": "2.29.1",
-    "eslint-plugin-jest-dom": "3.9.4",
-    "eslint-plugin-jsdoc": "44.2.7",
-    "eslint-plugin-jsx-a11y": "6.7.1",
-    "eslint-plugin-no-only-tests": "2.6.0",
+    "eslint-plugin-jest-dom": "5.1.0",
+    "eslint-plugin-jsdoc": "48.2.1",
+    "eslint-plugin-jsx-a11y": "6.6.0",
```
Comment (Nirajn2311) [inline package.json:132]: What's the reason behind downgrading this package?

**Comment by Sembauke** [comment]
No worries, I will update the pr accordingly.

**Review by huyenltnguyen — APPROVED** [review]

**Comment by raisedadead** [comment]
Looks good, except this warning: https://github.com/freeCodeCamp/freeCodeCamp/pull/54092/checks?check_run_id=22826866525

**Comment by Sembauke** [comment]
That is unfortunate, it is throwing a false positive.

File: client/src/templates/Challenges/dialogue/show.tsx [hunk 7]
```diff
@@ -233,8 +233,13 @@ class ShowDialogue extends Component<ShowDialogueProps, ShowDialogueState> {
                   <ChallengeHeading heading={t('learn.assignments')} />
                   <div className='video-quiz-options'>
                     {assignments.map((assignment, index) => (
-                      <label className='video-quiz-option-label' key={index}>
+                      <label
+                        className='video-quiz-option-label'
+                        htmlFor={`assignment-${index}`}
```
[the remark this answers is not in the document]
Comment (Sembauke) [reply client/src/templates/Challenges/dialogue/show.tsx:238]: I see, I am not much of accessibility person. I just read the same thing on their docs.

**Review by Nirajn2311 — APPROVED** [review]

**Pull request merged**